### PR TITLE
adds pref-link generation method in sdk

### PR DESCRIPTION
**Pull request by suhasdeshpande**
This PR adds a method that allows generating preference center URL that you can use to send your users to the Courier's hosted Preference Center to let them manage their notification preferences. You can use this URL in your application to send your users to the Preference Center.

```ts
const prefCenterLink = courier.generatePreferencesUrl("<your_user_id>", {
  // optional
  brandId: "<your_brand_id>",
});
```

### CODE REVIEW

File: packages/courier-js/src/helpers/client.ts [hunk 1]
```diff
@@ -26,7 +31,7 @@ async function tryCatch(
 export class Courier {
   private authorization?: string;
   private baseUrl: string;
-  private clientKey?: string;
+  public clientKey?: string;
```
Comment (suhasdeshpande) [inline packages/courier-js/src/helpers/client.ts]:
```suggestion
  private clientKey?: string;
```

File: packages/courier-js/src/helpers/client.ts [hunk 2]
```diff
@@ -96,4 +101,18 @@ export class Courier {
     };
     await tryCatch(deleteFn, this.debug);
   }
+
+  generatePreferencesUrl(
+    userId: string,
+    options?: PreferenceLinkOptions
+  ): string {
+    if (!userId || !this.clientKey) {
+      throw new Error("userId is required");
```
Comment (suhasdeshpande) [inline packages/courier-js/src/helpers/client.ts]:
```suggestion
      throw new Error("userId and clientKey are required");
```

File: packages/courier-js/package.json [hunk 3]
```diff
@@ -1,6 +1,6 @@
 {
   "name": "@trycourier/courier-js",
-  "version": "1.0.1",
+  "version": "1.1.0",
```
Comment (scarney81) [inline packages/courier-js/package.json]: Why are we manually versioning package.json?

File: packages/courier-js/src/index.ts [hunk 4]
```diff
@@ -15,7 +19,11 @@ const client = {
     if (!userId) {
       throw new Error("userId is required");
     }
-    await this.instance.post(`identify/${userId}`, payload);
+    await this.instance.post(`identify/${userId}`, {
+      profile: {
+        ...payload,
+      },
+    });
```
Comment (scarney81) [inline packages/courier-js/src/index.ts]: Why this change? This seems odd to include

Comment (suhasdeshpande) [reply packages/courier-js/src/index.ts]: I should open a separate PR for this. This was a regression introduced while changing `PUT` to `POST`
